vignettes/SEIRDV.Rmd

```diff
@@ -0,0 +1,215 @@
+becomes infectious to others. Infectious individuals can either recover or die
+because of the disease. Infectious individuals recover in $1/ \gamma$
+days, thus individuals recover at rate $\gamma$. Similarly, infectious
+individuals die due to the infection after $1/ \mu$ days, making the death rate $\mu$.
+Individuals are vaccinated at rate $\nu$. Among those who are immune (
+recovered and vaccinated), people lose their immunity against the virus at different
+rates depending on what type of immunity they acquired: $\delta_V$ for those vaccinated
+and $\delta_R$ respectively for those previously infected.
+
+In addition to these four main ODEs, the model also keeps track of the cumulative number 
+of cases ($C$) and the cumulative number of disease-related deaths ($D$) over time
+$$\frac{\text{d}C}{\text{d}t} = \beta S I,$$
+$$\frac{\text{d}D}{\text{d}t} = \mu I.$$
+
+The system is closed by specifying the initial conditions
+$$S(0) = S_0,\ E(0) = E_0,\ I(0) = I_0,\ R(0) = R_0,\ V(0) = V_0,\ C(0) = C_0,\ D(0) = D_0,$$
+where $C_0 = E_0 + I_0 + R_0$ to ensure the cases that occur before $t=0$ are
+also counted.
+In the implementation of the model in this package, the variables are normalized so 
+$S(t) + E(t) + I(t) + R(t) + V(t) + D(t) \equiv 1$ for any given $t$. This means that each of these state variables represents the fraction of the population in each of these states at a given point in time.
+
+We next illustrate how this model works using functionality available in `comomodels`.
+
+## The `SEIRDV` class
+
+We first create an `SEIRDV` object that we will use to run simulations.
+```{r}
+my_model <- SEIRDV()
+```
+
+Next, we set the parameter values for $\beta$, $\kappa$, $\gamma$, $\mu$, $\nu$,
+$\delta_V$ and $\delta_R$:
+```{r}
+params <- list(beta=1, kappa=0.9, gamma=0.5, mu=0.1, nu=0.4, delta_V=0.1,
+               delta_R=0.05)
+transmission_parameters(my_model) <- params
+```
+
+To check that the parameters have been set correctly, we call
+```{r}
+transmission_parameters(my_model)
+```
+
+We now need to set the initial conditions for the model. We only set the initial conditions $S(0)$, $E(0)$, $I(0)$, $R(0)$ and $V(0)$.
+```{r}
+initial_conditions(my_model) <- list(S0=0.99, E0=0, I0=0.01, R0=0, V0=0)
+```
+The initial conditions must sum to one. If they do not, an error is thrown.
+
+Now we simulate the system from $t=0$ days to $t=50$ days.
+```{r}
+times <- seq(0, 50, by = 1)
+out_df <- run(my_model, times)
+```
+
+The simulation returns two objects: one is a data frame comprising the states over time. Here, there's a peak in the infectious proportion between $t=15$ and $t=20$.
+```{r, fig.height = 5, fig.width = 7}
+states <- out_df$states
+
+ggplot(states, aes(x = time, y = value)) +
+  geom_line(aes(color = compartment)) +
+  scale_color_brewer(palette = "Dark2") +
+  labs(x = "time (days)", y = "fraction of the population",
+       title = glue("Evolution of different compartments over time"))
```

Comment:
It's hard to see E, I, R, and D on this figure because S and V are so much larger. Maybe have two panels, one for each set